front/src/modules/trainschedule/components/ImportTrainSchedule/ImportTrainScheduleConfig.tsx

```diff
@@ -203,13 +199,36 @@ const ImportTrainScheduleConfig = ({
         const formattedArrivalTime = `${startDate} ${arrivalTime}`;
         const formattedDepartureTime = `${startDate} ${departureTime}`;
 
+        let stopFor: number | undefined;
+
+        const arrivalDate = new Date(`${startDate}T${arrivalTime}`);
+        const departureDate = new Date(`${startDate}T${departureTime}`);
+        if (ocpType === 'stop') {
+          if (arrivalTime && departureTime) {
+            stopFor = Math.round((departureDate.getTime() - arrivalDate.getTime()) / 1000);
+          } else {
+            stopFor = 0;
+          }
+        } else if (ocpType === 'pass') {
+          if (arrivalTime && departureTime && arrivalTime !== departureTime) {
+            // Incorrect data, it shouldn't happen on a ocpType pass, skip step
+            return null;
+          }
+          if (isLastOcp) {
+            stopFor = 0;
+          } else {
+            stopFor = undefined;
+          }
+        }
+
         return {
           id: nextId(),
           uic,
           chCode,
           name: ocpRef,
-          arrivalTime: cleanTimeFormat(formattedArrivalTime),
-          departureTime: cleanTimeFormat(formattedDepartureTime),
+          arrivalTime: formattedArrivalTime,
+          departureTime: formattedDepartureTime,
+          duration: stopFor || undefined,
```

Comment:
`|| undefined` breaks zero duration stops: `0` is transformed to `undefined`, which has a different meaning.

Reply:
But there's no need to convert anything here, the original value is already a `number | undefined`.